pandas-stubs/core/series.pyi

```diff
@@ -216,13 +216,43 @@ class Series(IndexOpsMixin, NDFrame, Generic[S1]):
     @overload
     def __new__(
         cls,
-        data: IntervalIndex[IntervalT],
+        data: IntervalIndex[Interval[int]],
         index: Axes | None = ...,
         dtype=...,
         name: Hashable | None = ...,
         copy: bool = ...,
         fastpath: bool = ...,
-    ) -> Series[IntervalT]: ...
+    ) -> Series[Interval[int]]: ...
+    @overload
```

Comment:
might be nice to add a TODO referencing the mypy issue

Reply:
> might be nice to add a TODO referencing the mypy issue

I didn't research whether there exists a mypy issue, or how to create a small sample to replicate.  

I could add a comment indicating that we prefer the `Series[IntervalT]` notation, but that it doesn't seem to work and reference the original pandas-stubs issue.  Let me know if I should do that.

Reply:
Would something like the following work?

```py
T = TypeVar("T", float, int, Timestamp, Timedelta)

        data: IntervalIndex[Interval[T]],
        ....
    ) -> Series[Interval[T]]: ...
```

Reply:
> 

I can't see how this would work, because we have:
```python
class IntervalIndex(IntervalMixin, ExtensionIndex, Generic[IntervalT]):
```

Since `IntervalIndex` is generic, and `Interval` is generic, you can't make `IntervalIndex` depend on the "inner" generic type.

Reply:
Okay, sorry for the trouble.